core/shaders/vertex.js

```diff
@@ -0,0 +1,51 @@
+export const vertex = (defines) =>
```

Comment:
This is a nit and likely a matter of personal preference, but why not use a proper `export function vertex(defines)` here? I tend to use arrow functions as lambdas. Here however it's a full-on function and a function declaration would be more readable to me.